pkg/lib/api/build_policies.go

```diff
@@ -24,54 +24,63 @@ func buildMultipart(request Request) (additionalHeaders map[string]string, body
 	if ok {
 		f, ok := val.(util.JsonString)
 		if !ok {
-			return additionalHeaders, body, fmt.Errorf("file:filename should be a string")
+			return additionalHeaders, body, nil, fmt.Errorf("file:filename should be a string")
 		}
 		replaceFilename = &f
 	}
 
 	additionalHeaders["Content-Type"] = w.FormDataContentType()
-	for key, val := range request.Body.(map[string]any) {
 
+	createPart := func(key string, val any) error {
 		if key == "file:filename" {
-			continue
+			return nil
 		}
 
-		pathSpec, ok := val.(util.JsonString)
+		rawPathSpec, ok := val.(util.JsonString)
 		if !ok {
-			return additionalHeaders, body, fmt.Errorf("pathSpec should be a string")
+			return fmt.Errorf("pathSpec should be a string")
 		}
-		if !util.IsPathSpec(pathSpec) {
-			return additionalHeaders, body, fmt.Errorf("pathSpec %s is not valid", pathSpec)
+		pathSpec, err := util.ParsePathSpec(rawPathSpec)
+		if err != nil {
+			return fmt.Errorf("pathSpec %s is not valid: %w", rawPathSpec, err)
 		}
 
-		var err error
-
-		_, file, err := util.OpenFileOrUrl(pathSpec, request.ManifestDir)
+		file, err := util.OpenFileOrUrl(pathSpec.Path, request.ManifestDir)
 		if err != nil {
-			return additionalHeaders, nil, err
+			return err
 		}
 		defer file.Close()
 
 		var part io.Writer
 		if replaceFilename == nil {
-			part, err = w.CreateFormFile(key, pathSpec[1:])
+			part, err = w.CreateFormFile(key, pathSpec.Path)
 		} else {
 			part, err = w.CreateFormFile(key, *replaceFilename)
 		}
 		if err != nil {
-			return additionalHeaders, nil, err
+			return err
 		}
 		if _, err := io.Copy(part, file); err != nil {
-			return additionalHeaders, nil, err
+			return err
+		}
+
+		return nil
+	}
+
+	for key, val := range request.Body.(map[string]any) {
+		err = createPart(key, val)
+		if err != nil {
+			return additionalHeaders, body, bodyCloser, err
```

Comment:
Always return nil if err != nil for the other return values (unless its a struct)